packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/cap.dart

```diff
@@ -6,27 +6,56 @@ import 'package:flutter/foundation.dart' show immutable;
 
 import 'types.dart';
 
+/// Enumeration of possible types of caps.
+enum CapType {
+  /// Cap that is squared off exactly at the start or end vertex of a [Polyline]
+  /// with solid stroke pattern, equivalent to having no additional cap beyond
+  /// the start or end vertex.
+  butt,
+
+  /// Cap that is a semicircle with radius equal to half the stroke width,
+  /// centered at the start or end vertex of a [Polyline] with solid stroke
+  /// pattern.
+  round,
+
+  /// Cap that is squared off after extending half the stroke width beyond the
+  /// start or end vertex of a [Polyline] with solid stroke pattern.
+  square,
+
+  /// CustomCap with a bitmap overlay centered at the start or
+  /// end vertex of a [Polyline], orientated according to the direction of the line's
+  /// first or last edge and scaled with respect to the line's stroke width.
+  custom,
+}
+
+String _capTypeToJson(CapType capType) => switch (capType) {
```

Comment:
If there is documentation for where these json values come from please link it here.

Reply:
These are the old string values used before this PR's diff which are then used the platform packages' JSON-based approach. They do not correspond to any values outside this repo, e.g. the Maps SDK.

Reply:
Perfect that means if we change them then nothing bad happens.

Reply:
> Perfect that means if we change them then nothing bad happens.

No, that would be extremely bad. It would break all older versions of `google_maps_flutter`.

We're still cleaning up from tech debt mistakes in federating this plugin, where during federation magic string values that used to be internal to the monolithic plugin package were separated, with the Dart code producing them moved to `google_maps_flutter_platform_interface`, while the native code consuming them was left in the main package (and then eventually moved to platform implementation packages).

What *should* have happened was that we stopped using `toJson` from `platform_interface` as the serialization mechanism for the platform channels, in favor of package-internal versions. But since we didn't, these string values are effectively part of the public API surface of this package, consumed by our own implementations, with which we have to maintain compatibility.

Reply:
Ah ok so these strings while not needing to mirror something external like the google maps sdk they do need to mirror something external to the package `google_maps_flutter_platform_interface` for cross version compatibility. 

If we changed the values then set a new minimum for each of the platform implementations would that prevent a breakage? I am not asking because I want to do that but because I want to understand the relationships. 

The reason why I was asking what they needed to align to was to add documentation to warn future maintainers what those values needed to align with.

Reply:
> If we changed the values then set a new minimum for each of the platform implementations would that prevent a breakage?

No, because the versions of the platform implementations that call these `toJson` functions (and rely on the result matching what's in the native code) already exist, and people are using them. And those packages use standard `^2.x.y` constraints on `google_maps_flutter_platform_interface` and rely on the fact that doing so is safe because a change that breaks them wouldn't be made in a 2.x version.

Releasing a new `google_maps_flutter_android` would mean that people using the **new** `google_maps_flutter_android` would only get a compatible version of `google_maps_flutter_platform_interface`, but it would not stop people using an **old** version of `google_maps_flutter_android` from getting a new and incompatible version of `google_maps_flutter_platform_interface`.

> I am not asking because I want to do that but because I want to understand the relationships.

I think the easiest mental model is: because of the federation mistake, the exact structure and string values returned by these `toJson` functions are now, in practice, part of the official API surface of the method, and so changing any existing details of what it returns would be just like any incompatible change to the documented behavior of any public method: a breaking change requiring a major version bump.

From a docs standpoint, explicitly documenting what the return value structure is in the dart doc would accurately express the unfortunate state we are in now.